### ISSUE TITLE
Fix `core`'s inconsistencies

### ISSUE THREAD

**Issue by D4isDAVID** [discordjs/discord.js#9618]
### Which application or package is this feature request for?

core

### Feature

Currently API methods get added to different API classes based on three things:

1. Its topic on the Discord API docs, e.g. `WebhooksAPI#create` is part of the [Webhook](https://discord.com/developers/docs/resources/webhook#create-webhook) topic on the Discord docs, but uses the `/channels` route.
2. Its route on the Discord API, e.g. `ChannelsAPI#getWebhooks` uses the `/channels` route, but is part of the [Webhook](https://discord.com/developers/docs/resources/webhook#get-channel-webhooks) topic on the Discord docs.
3. Convenience, e.g. `ThreadsAPI` really just contains methods which use the Channels API, but having a different class can be more convenient when wording the methods, and as a result the parent class becomes smaller.

The problem is mostly with the first two, because notice how similar the two examples I gave are. They both use the `/channels` route, both appear in the Webhook topic, but somehow end up being in different API classes.

`ThreadsAPI` also has a bit of a problem:

- `ThreadsAPI#get` is literally the same as `ChannelsAPI#get` in terms of code after #8897, the only difference is that `ThreadsAPI#get` has a different return type, which isn't actually guaranteed (it doesn't necessarily return a thread, but a channel instead).
- `ThreadsAPI#create` and `ThreadsAPI#createForumThread` suffer from a problem similar but not exactly identical to the aforementioned one about routes and topics. This time they both appear in the same topic, route, and API class, but the action itself is done on a channel, not a thread, similar to `ChannelsAPI#createInvite` and the rest of the thread-related methods in `ChannelsAPI`. I guess you could say that the topic of the methods are the threads, but the actions themselves are done on channels.

In addition there's a minor problem where the documentation uses multiple different ways to word stuff such as parameters, for example:

- `The <parameter> for ...`
- `The <parameter> to use when ...`

### Ideal solution or implementation

I believe that adding methods to classes based on their route makes more sense.

When it comes to `ThreadsAPI`, I think its existence does make sense, but it still is just the Channels API and could be merged into `ChannelsAPI`.
Other than that, at least `ThreadsAPI#create`, and `ThreadsAPI#createForumThread` should be moved to `ChannelsAPI`, if we are going to choose to group methods by routes instead of topics.
`ThreadsAPI#get` should be completely removed. Or, instead, it may also check whether the channel is actually a thread or not, but I think the library users are the ones who should be doing that.

When it comes to the documentation, I think it should use the more concise phrasing.

### Other context

[#9578 (review)](https://github.com/discordjs/discord.js/pull/9578#pullrequestreview-1432883615)

I can make the PR for this, but I wanted to spark a conversation first to agree on a decision.

**Comment by suneettipirneni** [comment]
I agree with these changes. It would follow what the guilds api already does where other structures that relate to it are put in the same file.
